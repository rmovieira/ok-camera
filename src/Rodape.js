import React, { PureComponent } from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';
import Icon from 'react-native-vector-icons/Feather';

export default class Rodape extends PureComponent {
    render() {
        let botoes;
        if (this.props.emGravacao) {
            botoes = (
                <TouchableOpacity testID={'parar-gravacao'} onPress={this.props.pararGravacaoDeVideo} style={styles.capture}>
                    <Icon name="video-off" size={40} color="#900" />
                </TouchableOpacity>
            );
        } else {
            botoes = (
                <TouchableOpacity testID={'iniciar-gravacao'} onPress={this.props.iniciarGravacaoDeVideo} style={styles.capture}>
                    <Icon name="video" size={40} color="#900" />
                </TouchableOpacity>
            );
        }
        return (
            <View testID={'rodape'} onAnimationEnd={this.onAnimationEnd} animation={this.props.mostrarBotoes ? 'fadeIn' : 'fadeOut'} style={styles.painelBotoes} >
                <TouchableOpacity testID={'tirar-foto'} onPress={this.props.tirarFoto} style={styles.capture}>
                    <Icon name="camera" size={40} color="#900" />
                </TouchableOpacity>
                {botoes}
            </View>
        );
    }
}

const styles = StyleSheet.create({
    capture: {
        flex: 0,
        backgroundColor: '#fff',
        borderRadius: 5,
        alignSelf: 'center',
        padding: 10,
        margin: 20,
    },
    painelBotoes: { flex: 0, flexDirection: 'row', justifyContent: 'center' },
});