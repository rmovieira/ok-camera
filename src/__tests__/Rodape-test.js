import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Rodape from '../Rodape';

describe('Testes da tela de configurações', () => {

    test('Deve montar o componente corretamente quando está em gravação', async () => {
        const props = {
            emGravacao: true,
            pararGravacaoDeVideo: jest.fn(),
            iniciarGravacaoDeVideo: jest.fn(),
            mostrarBotoes: jest.fn(),
        }
        const rodape = shallow(<Rodape {...props} />);

        expect(rodape.find({ testID: 'iniciar-gravacao' })).not.toExist();
        expect(rodape.find({ testID: 'parar-gravacao' })).toExist();
        expect(rodape.find({ testID: 'tirar-foto' })).toExist();
        expect(toJson(rodape)).toMatchSnapshot();
    });

    test('Deve montar o componente corretamente quando não está em gravação', async () => {
        const props = {
            emGravacao: false,
            pararGravacaoDeVideo: jest.fn(),
            iniciarGravacaoDeVideo: jest.fn(),
            mostrarBotoes: jest.fn(),
        }
        const rodape = shallow(<Rodape {...props} />);

        expect(rodape.find({ testID: 'iniciar-gravacao' })).toExist();
        expect(rodape.find({ testID: 'parar-gravacao' })).not.toExist();
        expect(rodape.find({ testID: 'tirar-foto' })).toExist();
        expect(toJson(rodape)).toMatchSnapshot();
    });

    test('Deve chamar a função que para a gravação de vídeo ao pressionar o respectivo botão quando estiver em uma gravação', async () => {
        const props = {
            emGravacao: true,
            pararGravacaoDeVideo: jest.fn(),
            iniciarGravacaoDeVideo: jest.fn(),
            tirarFoto: jest.fn(),
            mostrarBotoes: true,
        }
        const rodape = shallow(<Rodape {...props} />);

        rodape.find({ testID: 'parar-gravacao' }).props().onPress();

        expect(props.pararGravacaoDeVideo).toHaveBeenCalledTimes(1);
    });

    test('Deve chamar a função que inicia a gravação de vídeo ao pressionar o respectivo botão quando não estiver em uma gravação', async () => {
        const props = {
            emGravacao: false,
            pararGravacaoDeVideo: jest.fn(),
            iniciarGravacaoDeVideo: jest.fn(),
            tirarFoto: jest.fn(),
            mostrarBotoes: true,
        }
        const rodape = shallow(<Rodape {...props} />);

        rodape.find({ testID: 'iniciar-gravacao' }).props().onPress();

        expect(props.iniciarGravacaoDeVideo).toHaveBeenCalledTimes(1);
    });


    test('Deve chamar a função que tira foto estando em gravação ou não', async () => {
        const props = {
            emGravacao: true,
            pararGravacaoDeVideo: jest.fn(),
            iniciarGravacaoDeVideo: jest.fn(),
            tirarFoto: jest.fn(),
            mostrarBotoes: true,
        }
        const rodape = shallow(<Rodape {...props} />);

        rodape.find({ testID: 'tirar-foto' }).props().onPress();

        expect(props.tirarFoto).toHaveBeenCalledTimes(1);

        rodape.setProps({ emGravacao: false });
        rodape.update();

        rodape.find({ testID: 'tirar-foto' }).props().onPress();

        expect(props.tirarFoto).toHaveBeenCalledTimes(2);
    });

});