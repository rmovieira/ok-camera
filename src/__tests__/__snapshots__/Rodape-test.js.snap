// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes da tela de configurações Deve montar o componente corretamente quando está em gravação 1`] = `
<View
  animation="fadeIn"
  style={
    Object {
      "flex": 0,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
  testID="rodape"
>
  <ForwardRef
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "flex": 0,
        "margin": 20,
        "padding": 10,
      }
    }
    testID="tirar-foto"
  >
    <Icon
      allowFontScaling={false}
      color="#900"
      name="camera"
      size={40}
    />
  </ForwardRef>
  <ForwardRef
    onPress={[MockFunction]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "flex": 0,
        "margin": 20,
        "padding": 10,
      }
    }
    testID="parar-gravacao"
  >
    <Icon
      allowFontScaling={false}
      color="#900"
      name="video-off"
      size={40}
    />
  </ForwardRef>
</View>
`;

exports[`Testes da tela de configurações Deve montar o componente corretamente quando não está em gravação 1`] = `
<View
  animation="fadeIn"
  style={
    Object {
      "flex": 0,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
  testID="rodape"
>
  <ForwardRef
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "flex": 0,
        "margin": 20,
        "padding": 10,
      }
    }
    testID="tirar-foto"
  >
    <Icon
      allowFontScaling={false}
      color="#900"
      name="camera"
      size={40}
    />
  </ForwardRef>
  <ForwardRef
    onPress={[MockFunction]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "flex": 0,
        "margin": 20,
        "padding": 10,
      }
    }
    testID="iniciar-gravacao"
  >
    <Icon
      allowFontScaling={false}
      color="#900"
      name="video"
      size={40}
    />
  </ForwardRef>
</View>
`;
